### PR TITLE
Add native iOS CLLocationManager tracking backend

### PR DESCRIPTION
**Pull request by tomasMizera**
Adds native `CLLocationManager`(https://developer.apple.com/documentation/corelocation/cllocationmanager?language=objc) implementation of position tracking backend. Used only on iOS devices (iPhones, iPads)

- supports background use
- notifies PositionTrackingManager about change in position. 
- uses `CLLocationManager distanceFilter` ( see https://developer.apple.com/documentation/corelocation/cllocationmanager/1423500-distancefilter?language=objc ) to update frequency of position updates
- does not use signal/slot connections as these are not delivered when app is minimised on iOS

### CODE REVIEW

File: app/position/tracking/iostrackingbackend.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+{
+  CoreUtils::log( TAG, QStringLiteral( "Error:" ) + QString::fromNSString( message ) );
+}
+
+void IOSTrackingBackend::stopTrying( NSString *message )
+{
+  CoreUtils::log( TAG, QStringLiteral( "Stopping updates, error:" ) + QString::fromNSString( message ) );
+
+  //
+  // let user know that tracking could not start/continue for some reason
```
Comment (PeterPetrik) [inline app/position/tracking/iostrackingbackend.cpp:60]: todo?

**Review by PeterPetrik — APPROVED** [review]
Am I right that on iOS the tracking is based on distance , but on desktop on time interval?

**Comment by tomasMizera** [comment]
> Am I right that on iOS the tracking is based on distance , but on desktop on time interval?

Yes, iOS API does not have an option to use time interval

File: app/position/tracking/iostrackingbackend.cpp (hunk 1 above)
Comment (tomasMizera) [reply app/position/tracking/iostrackingbackend.cpp:60]: Yeah, it is, but I skipped it for the initial implementation

**Pull request merged**